src/Combinators/parseGuards.ts

```diff
@@ -0,0 +1,36 @@
+import {checkerToGuard} from '../Checker';
+
+/**
+ * Would Number.parseFloat parse this input?
```

Comment:
shouldn't we have one that maps to `Number.parseInteger` too?

Reply:
And if we have that, I wonder if it should range check on the min/max safe int...

Reply:
ok, no, that can be done as a separate check, since +/- 1 to get out of bounds of MIN/MAX_SAFE_INTEGER parse as something outside those bounds

Reply:
It turns out that parseInteger is just parseFloat with a different return value (rounded off), as best I can tell, so it seemed pointless.

Range checks, if desired, can be done via combinators.

Reply:
> It turns out that parseInteger is just parseFloat with a different return value 
Well that's annoying...

There is a subtle difference between "is parseable as" vs. "is properly formatted as".  For the guards, though, I think the former is usually what we care about, the latter is more the responsibility of OAS type validation...

Reply:
I take that back. We don't have a combinator for "actually do the parsing then check further" -- I'm not even sure we should, since it adds a fair amount of complexity to what you're guarding for.